src/pexprs-assertAllApplicationsAreValid.js

```diff
@@ -64,8 +77,10 @@ pexprs.Apply.prototype._assertAllApplicationsAreValid = function(ruleName, gramm
     throw errors.undeclaredRule(this.ruleName, grammar.name, this.source);
   }
 
+  var isMeLexifiedContext = inLexifiedStackContext[inLexifiedStackContext.length - 1];
+  var isParLexifiedContext = inLexifiedStackContext[inLexifiedStackContext.length - 2];
   // ...and that this application is allowed
-  if (common.isSyntactic(this.ruleName) && (!common.isSyntactic(ruleName) || lexifyCount > 0)) {
+  if ((common.isSyntactic(this.ruleName) && isMeLexifiedContext) && (!common.isSyntactic(ruleName) || isParLexifiedContext)) {
```

Comment:
I don't get why you need to look at the 2nd value on the stack. Also, since you now initialize the stack with a correct initial value, you shouldn't need to check `common.isSyntactic(ruleName)`. I'd think it should just be this:

```
if (common.isSyntactic(this.ruleName) &&
    inLexifiedContextStack[inLexifiedContextStack.length - 1]) {
  // throw error
}
```